pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -178,6 +207,34 @@ func (r *ReferenceAdapter) checkRequirements() error {
 	return nil
 }
 
+// deleteProject checks the Project's lifecycle state of the projectReference.Spec.GCPProjectID instance in Google GCP
+// and deletes it if not active
+func (r *ReferenceAdapter) deleteProject() error {
+	project, err := r.gcpClient.GetProject(r.projectReference.Spec.GCPProjectID)
+	if err != nil {
+		return err
+	}
+
+	switch project.LifecycleState {
+	case "DELETE_REQUESTED":
+		r.logger.Info("Project lifecycleState == DELETE_REQUESTED")
+		return nil
+	case "LIFECYCLE_STATE_UNSPECIFIED":
+		r.logger.Error(operrors.ErrUnexpectedLifecycleState, "Unexpected LifecycleState", project.LifecycleState)
+		return operrors.ErrUnexpectedLifecycleState
+	case "ACTIVE":
+		r.logger.Info("Deleting Project")
+		_, err := r.gcpClient.DeleteProject(project.ProjectId)
+		if err != nil {
+			r.logger.Error(err, "could not delete project", project.Name, "with ProjectID", project.ProjectId)
+			return err
+		}
```

Comment:
is this logger output worth the extra if? The operator would log it anyhow. Without it's just return err :)